extensions/amp-story/1.0/media-pool.js

```diff
@@ -76,6 +76,13 @@ const POOL_MEDIA_ELEMENT_PROPERTY_NAME = '__AMP_MEDIA_POOL_ID__';
  */
 const ELEMENT_TASK_QUEUE_PROPERTY_NAME = '__AMP_MEDIA_ELEMENT_TASKS__';
 
+/**
+ * Single page ads may be injected later. If the original story contains 0 media
+ * elements the mediaPool will not be able to handle the injected audio/video
+ * Therefore we preallocate a minimum here.
+ * @const {number}
+ */
+const MINIMUM_MEDIA_ELEMENTS = 1;
```

Comment:
this number basically represents the size of media pool for pure text/image story.
I'm wondering if we can make this number bigger (say 2), to reduce the chance of evicting ad video

Reply:
Really good point!

How about, in ``amp-story.getMaxMediaElementCounts``, we replace

````
Math.min(videoMediaElementsCount, 8)
````

with

````
Math.min(videoMediaElementsCount + adVideoMediaElementsCount, 8)
````
where ``adVideoMediaElementsCount = 1``

That way we won't have to evict a video, unless the story has more than 7 videos. Basically that'd always save a spot for an ad video.

(and same logic for audio)

What do you think?

Reply:
Refactored. Moved logic to `getMaxMediaElementCounts` so that we add additional elements instead of just having a minimum. Also bumped the size up to 2